Automattic/jetpack
# Title Optimization: Add keywords section and make it send the keywords on the request

Fixes #38914.
Fixes #38917.

## Proposed changes:

* Add a keywords section to the UI, only visible when the beta features are enabled
* Make the button on this new section trigger a new generation and execute the request including a `keywords` property on the messages array
* Code refactoring to make things simpler

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Enable beta extensions, go to the block editor and add some paragraphs about something
* Open the development console on the browser and enable debug messages (`localStorage.setItem( 'debug', '*' )`)
* Look for the title optimization tool on the sidebar and click on the button to trigger it
* Confirm you see the "processing" state as usual
* Confirm the modal will show up correctly: 
   * the keywords section is visible at the top and looks like the proposed design
   * the generated suggestions are present on the body of the modal
   * the modal is wider than the current version (max of 700px instead of 512px)

<img width="600" alt="Screenshot 2024-08-15 at 18 28 58" src="https://github.com/user-attachments/assets/a85c9bb9-3bdf-4ca3-ae70-fda370336149">

* Confirm that a new generation will start when you click the "Generate again" button
* Confirm you can type something on the keywords textarea and click the "Generate again" button
* On the debug messages, filter for `dops:analytics` and confirm:
   * the tracking messages contain two new properties: `is_retry` and `has_keywords`
   * the `has_keywords` field is `false` when no keywords are typed, and `true` otherwise
   * the `is_retry` filed is generally `false`, but will turn to `true` when you are retrying the request after an error (you can force an error by setting the browser tab as offline before clicking the "Generate again" button)
* On the network activity tab of the browser, look for the requests to `jetpack-ai-query` and confirm the `keywords` property is set on the `context` object:

<img width="300" alt="Screenshot 2024-08-15 at 18 29 42" src="https://github.com/user-attachments/assets/571fe754-92b3-4806-9bb5-dc22656b352c">

* Confirm the new section goes away and the tool works as before, with no changes, when the beta extensions are disabled

Note: the provided keywords are still being ignored by the backend, so they will not affect the results
